plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -860,6 +853,14 @@ instance TypeError NoConstraintsErrMsg => Monoid (Opaque term rep) where
 
 -- Utils
 
+coerceVia :: Coercible a b => (a -> b) -> a -> b
+coerceVia _ = coerce
+{-# INLINE coerceVia #-}
+
+coerceArg :: Coercible a b => (a -> r) -> b -> r
```

Comment:
I very much do when it comes to performance-critical code. It's pretty common to call `coerce` over the entire function instead of just its argument, because that does not create any closures and in some cases makes the whole definition more sharing-friendly. I'll make a comment.